mods/default/nodes.lua

```diff
@@ -500,16 +500,39 @@ minetest.register_node("default:sapling", {
 	sunlight_propagates = true,
 	walkable = false,
 	on_timer = default.grow_sapling,
-	on_construct = function(pos)
-		minetest.get_node_timer(pos):start(math.random(2400,4800))
-	end,
 	selection_box = {
 		type = "fixed",
 		fixed = {-0.3, -0.5, -0.3, 0.3, 0.35, 0.3}
 	},
 	groups = {snappy = 2, dig_immediate = 3, flammable = 2,
 		attached_node = 1, sapling = 1},
 	sounds = default.node_sound_leaves_defaults(),
+
+	on_construct = function(pos)
+		minetest.get_node_timer(pos):start(math.random(2400,4800))
+	end,
+
+	on_place = function(itemstack, placer, pointed_thing)
+		-- Position of sapling
+		local pos = pointed_thing.above
+		local player_name = placer:get_player_name()
+		if not default.intersects_protection(
+					-- minp, maxp of resulting tree at maximum height
+					{x = pos.x - 2, y = pos.y, z = pos.z - 2},
+					{x = pos.x + 2, y = pos.y + 6, z = pos.z + 2},
+					player_name,
+					-- interval of interior volume check 
+					4
+				) then
+			minetest.set_node(pos, {name = "default:sapling"})
```

Comment:
This will likely replace any unprotected item, if it is `pointed_thing.above` and ignores any `buildable_to` `pointed_thing.under`